core_test.go

```diff
@@ -1363,7 +1363,7 @@ func benchmarkCoreBatch(b *testing.B, threads int) {
 	b.SetParallelism(threads)
 	b.RunParallel(func(pb *testing.PB) {
 		for pb.Next() {
-			if err = c.Batch(func(txn *Transaction) (err error) {
+			if err = c.Batch(context.Background(), func(txn *Transaction) (err error) {
```

Comment:
Do we have test for contexts handling failures?

Reply:
We should add that, good call

Reply:
The more I think about this, the more I like the idea @dhalman. I'm going to pull this off as a separate PR if you don't mind.